src/header/launchbar/BlastLaunchbarButton.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { useAppSelector } from 'src/store';
+
+import { getUnviewedBlastSubmissions } from 'src/content/app/tools/blast/state/blast-results/blastResultsSelectors';
+
+import LaunchbarButtonWithNotification from './LaunchbarButtonWithNotification';
+import { BlastIcon } from 'src/shared/components/app-icon';
+
+import styles from './Launchbar.scss';
+
+const BLAST_APP_ROOT_PATH = '/blast';
+
+const BlastLaunchbarButton = () => {
+  const unviewedSubmissions = useAppSelector(getUnviewedBlastSubmissions);
+  const location = useLocation();
+  const [blastAppPath, setBlastAppPath] = useState(BLAST_APP_ROOT_PATH);
```

Comment:
Ooh, clever! I didn't think we could pull this off without redux. Of course our hope is that the launchbar will never be unmounted over the lifetime of the page, i.e. that we will not ever have a screen that doesn't have the launchbar on it.